Allow to navigate by multiple years / issue #102

MatthieuBarthel opened the pull request:
Here is a first shot :

1. Ok, the step is configurable.
2. Ok, the feature is disabled by default.
3. Ok, the buttons behavior with min/max dates seems fine to me.
4. Not ok, I tried to add some e2e tests but I cannot trigger a click event on the fast/forward buttons, I get the message `Failed: element not visible`. I can't figure out what's going on, I see the buttons on the screen while running the tests but what ever I tried it fails. Could you help ?

Waiting for your feedback, don't hesitate if you want me to do some update, like renaming variables :-)

Thank you !

vlio20 commented on src/app/calendar-nav/calendar-nav.component.less:
```diff
@@ -16,11 +16,25 @@
+    right: 5px;
+  }
+
+  .dp-calendar-nav-left, .dp-calendar-nav-right, .dp-calendar-secondary-nav-left, .dp-calendar-secondary-nav-right {
```
please break to lines

vlio20 commented on src/app/calendar-nav/calendar-nav.component.less:
```diff
@@ -30,11 +44,15 @@
-  .dp-calendar-nav-right {
-    right: 5px;
+  .dp-calendar-secondary-nav-left::before, .dp-calendar-secondary-nav-right::before {
+    content: "«";
```
Icons should be made with pure CSS in order to support different OS

vlio20 commented on src/app/calendar-nav/calendar-nav.component.less:
```diff
@@ -30,11 +44,15 @@
+  }
+
+  .dp-calendar-secondary-nav-right::before {
+    content: "»";
```
same here

vlio20 commented on src/app/calendar-nav/calendar-nav.component.less:
```diff
@@ -55,7 +73,7 @@
     }
 
-    .dp-calendar-nav-left, .dp-calendar-nav-right {
+    .dp-calendar-nav-left, .dp-calendar-nav-right, .dp-calendar-secondary-nav-left, .dp-calendar-secondary-nav-right {
```
break to lines pls

vlio20 commented on src/app/month-calendar/month-calendar.component.ts:
```diff
@@ -163,19 +163,45 @@ export class MonthCalendarComponent implements OnInit, OnChanges, ControlValueAc
 
+  onRightSecondaryNav() {
+    let navigateBy = this.componentConfig.multipleYearsNavigateBy
+    if (this.componentConfig.max && this.componentConfig.max.year() - this.currentDateView.year() < navigateBy) {
```
lets put this if logic into a variable

vlio20 commented:
@MatthieuBarthel, are you planning to fix the comments?

MatthieuBarthel commented:
Thank you for your feedback and sorry for the late answer. I just pushed an update :
1. I first pushed with the x-prefix, tests went fine. As I had many tests failing locally, I tried to remove the prefix, but it failed: https://travis-ci.org/vlio20/angular-datepicker/builds/259683865 
2. What is missing ? I updated the README and demo.
3. Fixed.

Let me know :-)

vlio20 commented on src/app/calendar-nav/calendar-nav.component.html:
```diff
@@ -1,21 +1,37 @@
+  <div class="dp-calendar-nav-container-right">
+    <button type="button"
+            class="dp-calendar-secondary-nav-right"
+            [hidden]="!showRightSecondaryNav"
```
why not to use *ngIf

vlio20 commented on src/app/common/styles/variables.less:
```diff
@@ -20,3 +20,21 @@
   }
 }
+
+.double-arrow(@deg) {
```
can you please post an image of this?

MatthieuBarthel commented:
Here are the double arrows :

![image](https://user-images.githubusercontent.com/435534/28863846-c06f3e80-776a-11e7-9326-befd2cfab4c8.png)
